File: app/src/main/kotlin/com/yuyakaido/android/flow/di/component/AppComponent.java
package com.yuyakaido.android.flow.di.component;

import com.yuyakaido.android.flow.di.module.AppModule;
import com.yuyakaido.android.flow.presentation.di.PresentationComponent;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by yuyakaido on 7/30/16.
 */
@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    PresentationComponent newPresentationComponent();
}

File: app/src/main/kotlin/com/yuyakaido/android/flow/di/component/AppComponent.kt
package com.yuyakaido.android.flow.di.component

import com.yuyakaido.android.flow.di.module.AppModule
import com.yuyakaido.android.flow.di.module.ArticleListModule
import com.yuyakaido.android.flow.di.module.GetCategoryModule
import dagger.Component
import javax.inject.Singleton

/**
 * Created by yuyakaido on 7/30/16.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun newCategoryPagerComponent(module: GetCategoryModule): CategoryPagerComponent
    fun newArticleListComponent(module: ArticleListModule): ArticleListComponent
    fun newQiitaTagComponent(): QiitaTagComponent
    fun newQiitaPostComponent(): QiitaPostComponent
}
